core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -140,7 +140,9 @@ private boolean runUpxInContainer(NativeImageBuildItem nativeImage, NativeConfig
                 volumeOutputPath + ":" + NativeImageBuildStep.CONTAINER_BUILD_VOLUME_PATH + ":z");
 
         commandLine.add(effectiveBuilderImage);
-        commandLine.add(level);
+        if (nativeConfig.compression().level().isPresent()) {
+            commandLine.add(getCompressionLevel(nativeConfig.compression().level().getAsInt()));
```

Comment:
I didn't know you could run without a compression level.

Reply:
According to the docs, "The default compression level is -8 for files smaller than 512 KiB, and -7 otherwise". Since the property is an `OptionalInt`, I took that to mean that the property is optional, and if not given . So, my assumption is that this was expected to work this way, but I was not able to test it locally.